### PR TITLE
Add configurable damage and defense parameters

### PR DESCRIPTION
**Pull request by dydzio0614**
Reimplementation of https://github.com/vcmi/vcmi/pull/435

### CODE REVIEW

**Comment by IvanSavenko** [comment]
#435 should be closed then?
Also, your old code had serialization compatibility code while your current one doesn't

**Comment by dydzio0614** [comment]
Yep, #435 to be closed. I saw serialization compatibility code got removed for hardcoded features so I kept it removed

File: lib/battle/CBattleInfoCallback.cpp [hunk 1]
```diff
@@ -802,12 +803,18 @@ TDmgRange CBattleInfoCallback::calculateDmgRange(const BattleAttackInfo & info)
 	//bonus from attack/defense skills
 	if(attackDefenceDifference < 0) //decreasing dmg
 	{
-		const double dec = std::min(0.025 * (-attackDefenceDifference), 0.7);
+		double defenseMultiplier = VLC->modh->settings.DEFENSE_POINT_DMG_MULTIPLIER;
+		double defenseMultiplierCap = VLC->modh->settings.DEFENSE_POINTS_DMG_MULTIPLIER_CAP;
```
Comment (Nordsoft91) [inline lib/battle/CBattleInfoCallback.cpp:807]:
```suggestion
		const double defenseMultiplier = VLC->modh->settings.DEFENSE_POINT_DMG_MULTIPLIER;
		const double defenseMultiplierCap = VLC->modh->settings.DEFENSE_POINTS_DMG_MULTIPLIER_CAP;
```

File: config/defaultMods.json [hunk 2]
```diff
@@ -26,7 +26,11 @@
 		"MAX_HEROES_ON_MAP_PER_PLAYER" : 8,
 		"WINNING_HERO_WITH_NO_TROOPS_RETREATS": true,
 		"BLACK_MARKET_MONTHLY_ARTIFACTS_CHANGE": true,
-		"NO_RANDOM_SPECIAL_WEEKS_AND_MONTHS": false
+		"NO_RANDOM_SPECIAL_WEEKS_AND_MONTHS": false,
+		"ATTACK_POINT_DMG_MULTIPLIER": 0.05,
+		"ATTACK_POINTS_DMG_MULTIPLIER_CAP": 4.0,
```
Comment (Nordsoft91) [inline config/defaultMods.json:31]: what does CAP mean?

File: lib/battle/CBattleInfoCallback.cpp [hunk 3]
```diff
@@ -802,12 +803,18 @@ TDmgRange CBattleInfoCallback::calculateDmgRange(const BattleAttackInfo & info)
...
+
+		const double dec = std::min(defenseMultiplier * (-attackDefenceDifference), defenseMultiplierCap);
 		multBonus *= 1.0 - dec;
 	}
 	else //increasing dmg
 	{
-		const double inc = std::min(0.05 * attackDefenceDifference, 4.0);
+		double attackMultiplier = VLC->modh->settings.ATTACK_POINT_DMG_MULTIPLIER;
+		double attackMultiplierCap = VLC->modh->settings.ATTACK_POINTS_DMG_MULTIPLIER_CAP;
```
Comment (Nordsoft91) [inline lib/battle/CBattleInfoCallback.cpp:815]:
```suggestion
		const double attackMultiplier = VLC->modh->settings.ATTACK_POINT_DMG_MULTIPLIER;
		const double attackMultiplierCap = VLC->modh->settings.ATTACK_POINTS_DMG_MULTIPLIER_CAP;
```

**Review by Nordsoft91 — APPROVED** [review]

File: config/defaultMods.json (hunk 2 above)
Comment (kambala-decapitator) [reply config/defaultMods.json:31]: cap = maximum/limit. "damage cap" is a widely used term for example.

**Comment by IvanSavenko** [comment]
Merging tomorrow if there are no objection

**Pull request merged**